strutil/escape.go

```diff
@@ -0,0 +1,66 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+const allowed = `:_.abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789`
+
+// HexEscapePath works like systemd-escape --path
```

Comment:
There are a few corner cases where `HexEscapePath()` and `systemd-escape --path` behave differently, but unsure if this difference is relevant in this case. Examples: `.`, `a/..`, `a/../b`, `a/../../b` (generally `systemd-escape --path` returns an error for anything that goes back to `.`).

Reply:
`systemd-escape` is really picky about dots. I looked at the use cases in current code, and the only one is mount unit names for snap mount points. Those paths should be safe and dot-less.